pallets/automation-time/src/fees.rs

```diff
@@ -148,3 +148,78 @@ where
 		Ok(())
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::{mock::*, Action};
+	use frame_benchmarking::frame_support::assert_err;
+	use frame_support::sp_runtime::AccountId32;
+
+	#[test]
+	fn executes_prereq_function() {
+		new_test_ext(0).execute_with(|| {
+			let alice = AccountId32::new(ALICE);
+			get_funds(alice.clone());
+			let mut spy = 0;
+			let result = <Test as crate::Config>::FeeHandler::pay_checked_fees_for(
+				&alice,
+				&Action::Notify { message: vec![0] },
+				1,
+				|| {
+					spy += 1;
+					Ok("called")
+				},
+			);
+			assert_eq!(result.expect("success"), "called");
+			assert_eq!(spy, 1);
+		})
+	}
+
+	#[test]
+	fn errors_when_not_enough_funds_for_fee() {
+		new_test_ext(0).execute_with(|| {
+			let alice = AccountId32::new(ALICE);
+			let result = <Test as crate::Config>::FeeHandler::pay_checked_fees_for(
+				&alice,
+				&Action::Notify { message: vec![0] },
+				1,
+				|| Ok(()),
+			);
+			assert_err!(result, Error::<Test>::InsufficientBalance);
+		})
+	}
+
+	#[test]
+	fn charges_fees() {
+		new_test_ext(0).execute_with(|| {
+			let alice = AccountId32::new(ALICE);
+			get_funds(alice.clone());
+			let starting_funds = Balances::free_balance(alice.clone());
+			let _ = <Test as crate::Config>::FeeHandler::pay_checked_fees_for(
+				&alice,
+				&Action::Notify { message: vec![0] },
+				1,
+				|| Ok(()),
+			);
+			assert!(starting_funds > Balances::free_balance(alice))
```

Comment:
could this test be combined with executes_prereq_function? like basically having 1 success test case and we check the state of fee and prereq trigger after that